sklearn/metric_learning/nca.py

```diff
@@ -0,0 +1,216 @@
+from ..base import BaseEstimator, TransformerMixin
+
+import numpy as np
+import scipy as sp
+import scipy.misc
+import scipy.optimize
+import time
+
+import sys
+
+def nca_vectorized_oracle(X, y, params):
+    dX = X[:, None, :] - X[None, :, :]  # n_samples x n_samples x n_comp
+    outer = dX[:, :, None, :] * dX[:, :, :, None]  # n_samples x n_samples x n_comp x n_comp
```

Comment:
please use `np.newaxis` instead of `None`. What does `n_comp` stand for? Could you use `n_features` instead?